Use cloudinary.v2.uploader.explicit instead of cloudinary.v2.api.resource

eexit opened the pull request:
Hello Marco,

I'm recently trying to migrate all my images to Cloudinary so I'm diving into your code to see how it works. First of all, thank you for your work here.

Here a small PR to improve the behavior of `exists`:

- Returns explicit a boolean (to follow Ghost storage interface)
- Use `cloudinary.v2.uploader.explicit` because it's not rate limited as opposed as `cloudinary.v2.api.resource`

> Note that if you only need details about the original resource, you can also use the upload or explicit methods, which are not rate limited.

PS: you find a better way to review the PR with the [?w=1 param](https://github.com/mmornati/ghost-cloudinary-store/pull/12/files?w=1).

mmornati commented on index.js:
```diff
@@ -14,9 +14,10 @@ class CloudinaryAdapter extends BaseAdapter{
+      cloudinary.v2.uploader.explicit(path.parse(filename).name, {type: 'upload'}, function(error, result) {
         if (result) {
-          resolve(result);
+          resolve(true);
```
We don't need the result here (for ghost)? With only the true  resolve are we able to link the image to the post?

eexit replied:
The image URL in the post is returned by the `save` method. The `exists` method only returns a boolean.
However, I could just return the boolean without the resolve.

eexit commented:
@mmornati I just pushed an almost complete re-write of the adapter. I added some more tests and added support for image upload into folder.
https://github.com/eexit/ghost-storage-cloudinary

mmornati commented:
@eexit good job in all your refactor. Can I integrate all of this in mine one too?
